### PR TITLE
Math redone

### PR DESCRIPTION
**Pull request by b62303411**

### CODE REVIEW

File: gwave.hlsl [hunk 1]
```diff
@@ -1,7 +1,9 @@
 #ifndef GWAVEHLSLINCLUDE_INCLUDED
 #define GWAVEHLSLINCLUDE_INCLUDED
 
-float3  gwave(float3 vertex, float3 direction, float wl, float s, inout float3 tangent, inout float3 binormal)
+
+
+float3  gerstnerwave(float3 vertex, float3 direction, float wl, float s, inout float3 tangent, inout float3 binormal)
 {
 	float gravity = 9.81;
```
Comment (casselmanscott) [inline gwave.hlsl:8]: Suggest to make gravity precise as well (to be concise with PI):

9.80665

File: gwave_math.hlsl [hunk 2]
```diff
@@ -0,0 +1,132 @@
+#ifndef GWAVEHLSLINCLUDE_INCLUDED
+#define GWAVEHLSLINCLUDE_INCLUDED
+
+
+
+float CalculateWaveNumber(float wavelength)
+{
+	float g = 9.81;
```
Comment (casselmanscott) [inline gwave_math.hlsl]: Suggest to make gravity precise as well (to be concise with PI):

9.80665

File: gwave_math.hlsl [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	float pi = 3.14159265358979323846;
+	return 2 * pi / wavelength;
+}
+
+float CalculateFrequency(float wavelength, float steepness)
+{
+	float g = 9.81;
```
Comment (casselmanscott) [inline gwave_math.hlsl]: Suggest to make gravity precise as well (to be concise with PI):

9.80665

File: gwave_math.hlsl [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+float CalculatePeriod(float frequency)
+{
+	return 1 / frequency;
+}
+
+// Calculate the amplitude of a wave given its steepness and wavelength
+float CalculateAmplitude(float steepness, float wavelength)
+{
+	float g = 9.81;
```
Comment (casselmanscott) [inline gwave_math.hlsl]: Suggest to make gravity precise as well (to be concise with PI):

9.80665

File: gwave_math.hlsl [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+	// Calculate the amplitude of the wave using the formula provided above
+	float amplitude = steepness / wavenumber;
+
+	// Return the calculated amplitude
+	return amplitude;
+}
+
+float CalculateAngularFrequency(float period)
+{
+	float g = 9.81;
```
Comment (casselmanscott) [inline gwave_math.hlsl]: Suggest to make gravity precise as well (to be concise with PI):

9.80665

File: gwave_math.hlsl [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+	float pi = 3.14159265358979323846;
+	return 2 * pi / period;
+}
+
+void GerstnerWave(float3 position, float3 direction, float wavelength, float steepness, inout float3 displacement, inout float3 tangent, inout float3 binormal)
+{
+	float g = 9.81;
```
Comment (casselmanscott) [inline gwave_math.hlsl]: Suggest to make gravity precise as well (to be concise with PI):

9.80665

**Review by casselmanscott — CHANGES_REQUESTED** [review]

File: gwave_math.hlsl [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+	float amplitude = CalculateAmplitude(steepness, wavelength);
+	
+	float km = wavenumber;
+
+	displacement.x +=  - direction.x * amplitude * sin(theta);
+	displacement.y += amplitude * cos(theta);
+	displacement.z +=  - direction.z * amplitude * sin(theta);
+	
+    // ds/dx partial derivative for the tengeant.
+    tangent.x += - (amplitude* k.x * k.x * cos(theta))/km;
```
Comment (casselmanscott) [inline gwave_math.hlsl]: Fix indentation

**Review by casselmanscott — CHANGES_REQUESTED** [review]

**Review by casselmanscott — APPROVED** [review]
Approved

File: gwave_math.hlsl (hunk 3 above)
Comment (b62303411) [reply gwave_math.hlsl]: done

**Pull request merged**